mlflow/utils/autologging_utils.py

```diff
@@ -70,39 +36,31 @@ def log_fn_args_as_params(fn, args, kwargs, unlogged=[]):  # pylint: disable=W01
     :param unlogged: parameters not to be logged
     :return: None
     """
-    # all_default_values has length n, corresponding to values of the
-    # last n elements in all_param_names
-    pos_params, _, _, pos_defaults, kw_params, kw_defaults, _ = inspect.getfullargspec(fn)
-
-    kw_params = list(kw_params) if kw_params else []
-    pos_defaults = list(pos_defaults) if pos_defaults else []
-    all_param_names = pos_params + kw_params
-    all_default_values = pos_defaults + [kw_defaults[param] for param in kw_params]
-
-    # Checking if default values are present for logging. Known bug that getargspec will return an
-    # empty argspec for certain functions, despite the functions having an argspec.
-    if all_default_values is not None and len(all_default_values) > 0:
-        # Logging the default arguments not passed by the user
-        defaults = get_unspecified_default_args(args, kwargs, all_param_names, all_default_values)
-
-        for name in [name for name in defaults.keys() if name in unlogged]:
-            del defaults[name]
-        try_mlflow_log(mlflow.log_params, defaults)
-
-    # Logging the arguments passed by the user
-    args_dict = dict(
-        (param_name, param_val)
-        for param_name, param_val in zip(all_param_names, args)
-        if param_name not in unlogged
+    param_spec = inspect.signature(fn).parameters
+    # Filter out `self` from the signature under the assumption that it is not specified
+    # as an explicit parameter in the call to `fn`
+    relevant_params = [param for param in param_spec.values() if param.name != "self"]
```

Comment:
Not a huge fan of this change (i.e. what if `self` is actually the first element of `args`?). Will revisit this tomorrow.

Reply:
Seems acceptable - callers can filter out `self` via the patched function signature. I've documented the expectation that `self` is not a part of the `args` passed to this function.